warc_metadata_sidecar.py

```diff
@@ -202,27 +197,20 @@ def metadata_sidecar(archive_dir, warc_file, operator=None, publisher=None):
 
             print(url)
             payload.seek(0)
-            mime_and_puid = find_mime_and_puid(fido, payload)
+            mime_dict, puid = find_mime_and_puid(fido, payload)
             result_dict = {}
             lang_cld = None
-            if mime_and_puid[0] is not {}:
-                if mime_and_puid[0].get('fido') and mime_and_puid[0].get('python-magic'):
-                    record_mime = "r'(" + mime_and_puid[0].get('fido') + "|" + mime_and_puid[0].get('python-magic') + ")'"
-                else:
-                    if mime_and_puid[0].get('fido'):
-                        record_mime = mime_and_puid[0].get('fido')
-                    else:
-                        record_mime = mime_and_puid[0].get('python-magic')
-                # Check that the format matches either mime type or just the one.
-                if TEXT_FORMAT_MIMES.search(record_mime):
+            if mime_dict:
```

Comment:
I think this `if mime_dict` may not be needed because I think the `if TEXT_FORMAT_MIMES.search(' '.join(mime_dict.values()))` would be a falsy `''` (empty string) if `mime_dict` is empty. You should verify that thought  though.